ddtrace/context.py

```diff
@@ -69,6 +71,30 @@ def sampling_priority(self, value):
         with self._lock:
             self._sampling_priority = value
 
+    def set_ctx_item(self, key, val):
```

Comment:
Any reason not to use `__getitem__` and `__setitem__`?

Reply:
Yeah I debated going this approach. I guess it makes sense.